packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx

```diff
@@ -53,6 +54,50 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
     }
   };
 
+  /**
+   * the api will return sorted results and those are stored in state, when user
+   * changes the filter we dont make a new api call, and use the state. New data is
+   * fetched from api when user has reached the end of page.
+   * ---
+   * This function is responsible for sorting threads in state that were earlier
+   * sorted by another featured flag
+   */
+  const sortByFeaturedFilter = (t: Thread[]) => {
+    if (featuredFilter === 'oldest') {
```

Comment:
we should use here (and below) types or import `featuredFilterQueryMap` for comparisions

Reply:
[Resolved](https://github.com/hicommonwealth/commonwealth/pull/3926/commits/91746fe421d6d8615e2cf9676ed0cf323470f6ab)